community/cypher/cypher-compiler-3.2/src/main/scala/org/neo4j/cypher/internal/compiler/v3_2/planner/CostBasedExecutablePlanBuilder.scala

```diff
@@ -44,38 +43,27 @@ case class CostBasedExecutablePlanBuilder(monitors: Monitors,
                                           queryPlanner: QueryPlanner,
                                           queryGraphSolver: QueryGraphSolver,
                                           rewriterSequencer: (String) => RewriterStepSequencer,
-                                          semanticChecker: SemanticChecker,
                                           plannerName: CostBasedPlannerName,
                                           runtimeBuilder: RuntimeBuilder,
                                           updateStrategy: UpdateStrategy,
                                           config: CypherCompilerConfiguration,
                                           publicTypeConverter: Any => Any)
   extends ExecutablePlanBuilder {
 
-  override def producePlan(inputQuery: PreparedQuerySemantics, planContext: PlanContext, tracer: CompilationPhaseTracer,
+  override def producePlan(inputQuery: CompilationState, planContext: PlanContext, tracer: CompilationPhaseTracer,
                            createFingerprintReference: (Option[PlanFingerprint]) => PlanFingerprintReference) = {
-    val statement =
-      CostBasedExecutablePlanBuilder.rewriteStatement(
-        statement = inputQuery.statement,
-        scopeTree = inputQuery.scopeTree,
-        semanticTable = inputQuery.semanticTable,
-        rewriterSequencer = rewriterSequencer,
-        preConditions = inputQuery.conditions,
-        monitor = monitors.newMonitor[AstRewritingMonitor](),
-        semanticChecker = semanticChecker)
-
     //monitor success of compilation
     val planBuilderMonitor = monitors.newMonitor[NewRuntimeSuccessRateMonitor](CypherCompilerFactory.monitorTag)
 
-    statement match {
-      case (ast: Query, rewrittenSemanticTable) =>
+    inputQuery.statement match {
+      case ast: Query =>
         val (periodicCommit, logicalPlan, pipeBuildContext) = closing(tracer.beginPhase(LOGICAL_PLANNING)) {
-          produceLogicalPlan(ast, rewrittenSemanticTable)(planContext, planContext.notificationLogger())
+          produceLogicalPlan(ast, inputQuery.semanticTable)(planContext, planContext.notificationLogger())
         }
-          runtimeBuilder(periodicCommit, logicalPlan, pipeBuildContext, planContext, tracer, rewrittenSemanticTable,
+          runtimeBuilder(periodicCommit, logicalPlan, pipeBuildContext, planContext, tracer, inputQuery.semanticTable,
                          planBuilderMonitor, plannerName, inputQuery, createFingerprintReference, config)
       case x =>
-        throw new CantHandleQueryException(x.toString())
+        throw new InternalException(s"Can't plan a $x query with the cost planner")
```

Comment:
Shouldn't we keep the CantHandleQueryException to try fallback on 3.1 rule planner?